### PR TITLE
rose edit: macros: allow importing other modules from same directory

### PR DESCRIPTION
**Pull request by benfitzpatrick**
When a macro imports another module from the same directory as itself, the import
will fail in `rose edit` but succeed in `rose macro`, due to a slightly different treatment of
the module names (via the variable `as_name` in the code).

When importing a macro, we should add its base directory to the PATH.

@matthewrmshin, please review.
@r-sharp and @scwhitehouse, please track this issue.

### CODE REVIEW

File: lib/python/rose/macro.py [hunk 1]
```diff
@@ -578,6 +579,7 @@ def load_meta_macro_modules(meta_files, module_prefix=None):
         except Exception:
             rose.reporter.Reporter()(
                 MacroLoadError(meta_file, traceback.format_exc()))
+        sys.path.pop()
```
Comment (matthewrmshin) [inline lib/python/rose/macro.py]: I guess you are confident that you don't need to do this within a `finally` here?

Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Yes, I think it's fine - anything uncaught (a `BaseException`) would kill everything anyway.

**Comment by matthewrmshin** [comment]
@r-sharp or @scwhitehouse please sanity check.

File: lib/python/rose/macro.py [hunk 2]
```diff
@@ -568,6 +568,7 @@ def load_meta_macro_modules(meta_files, module_prefix=None):
         if (not meta_dir.endswith(MACRO_DIRNAME) or
                 not meta_file.endswith(MACRO_EXT)):
             continue
+        sys.path.append(meta_dir)
```
Comment (matthewrmshin) [inline lib/python/rose/macro.py]: Just wondering if it would be safer to prepend instead of append?

Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Yes, you are probably right.

**Comment by scwhitehouse** [comment]
This looks sensible to me; Roddy will be back at work on Monday.

**Comment by r-sharp** [comment]
Looks fine to me. Especially now Ben has made it an insert instead of an append.

**Pull request merged**